include/vinecopulib/misc/implementation/tools_optimization.ipp

```diff
@@ -46,6 +37,91 @@ inline void Optimizer::set_controls(double initial_trust_region,
                                maxeval);
 }
 
+//! @brief solve the maximization problem.
+//!
+//! @param initial_parameters of starting values for the optimization
+//!     algorithm.
+//! @param lower_bounds lower bounds for the parameters.
+//! @param upper_bounds upper bounds for the parameters.
+//! @param the objective function to maximize.
+//! @return the optimal parameters.
+inline Eigen::VectorXd Optimizer::optimize(
+    const Eigen::VectorXd &initial_parameters,
+    const Eigen::VectorXd &lower_bounds,
+    const Eigen::VectorXd &upper_bounds,
+    std::function<double(const Eigen::VectorXd&)> objective)
+{
+    check_parameters_size(initial_parameters, lower_bounds, upper_bounds);
+    size_t n_parameters = initial_parameters.size();
+    auto optimal_parameters = initial_parameters;
+    if (n_parameters > 1) {
+        //const int number_interpolation_conditions = (n_parameters + 1) *
+        //        (n_parameters + 2)/2;
+        int number_interpolation_conditions = n_parameters + 3;
```

Comment:
?

Reply:
I did not change this part!

Reply:
Haha, so I have no idea why I did that ^^